mbq/client/tests/test_client.py

```diff
@@ -1,23 +1,47 @@
 from unittest import TestCase
+from unittest.mock import Mock
 
 from .compat import patch
 from mbq.client.client import ServiceClient
 
 
 class CorrelationIDTestCase(TestCase):
-    def setUp(self):
-        self.client = ServiceClient('https://foo.com/')
 
-    def test_correlation_id_provided(self):
+    def test_correlation_id_getter_provided(self):
+        self.client = ServiceClient(
+            'https://foo.com/',
+            correlation_id_getter=lambda: 'hello-world'
+        )
         with patch('requests.get') as requests_mock:
-            self.client.get('/url', correlation_id='hello-world')
+            self.client.get('/url')
             self.assertEquals(1, requests_mock.call_count)
             headers = requests_mock.call_args[1]['headers']
             self.assertEquals('hello-world', headers['X-Correlation-Id'])
 
-    def test_correlation_id_not_provided(self):
+    def test_correlation_id_getter_not_provided(self):
+        self.client = ServiceClient('https://foo.com/')
         with patch('requests.get') as requests_mock:
             self.client.get('/url')
             self.assertEquals(1, requests_mock.call_count)
             headers = requests_mock.call_args[1]['headers']
-            self.assertTrue(headers['X-Correlation-Id'])
+            self.assertIsNone(headers.get('X-Correlation-Id'))
+
+    def test_multivalued_correlation_id_getter_provided(self):
+        cid_getter_mock = Mock()
+        cid_getter_mock.side_effect = ['hello-world', 'goodbye-world']
+        self.client = ServiceClient(
+            'https://foo.com/',
+            correlation_id_getter=cid_getter_mock
+        )
+
+        with patch('requests.get') as requests_mock:
+            self.client.get('/url')
+            self.assertEquals(1, requests_mock.call_count)
+            headers = requests_mock.call_args[1]['headers']
+            self.assertEquals('hello-world',headers.get('X-Correlation-Id'))
+
+            self.client.get('/url')
+            self.assertEquals(2, requests_mock.call_count)
+            headers = requests_mock.call_args[1]['headers']
+            self.assertEquals('goodbye-world',headers.get('X-Correlation-Id'))
+
```

Comment:
Please set your editor up to end every file with exactly one newline

Reply:
for lint?

Reply:
Linter may not check for it but it's considered good practice in my pretty much every language. Otherwise you just have dangling cruft at the end of your files